src/request.spec.ts

```diff
@@ -482,375 +367,282 @@ describe('Request', () => {
-        done(err);
       });
+    assert.strictEqual(timeout, retryAfter);
   });
 
-  it('repeats 503-throttled request until it succeeds', (done) => {
+  it('repeats 503-throttled request until it succeeds', async () => {
     let i: number = 0;
 
     sinon.stub(_request as any, 'req').callsFake(() => {
       if (i++ < 3) {
-        return Promise.reject({
+        throw {
           response: {
             status: 503,
             headers: {}
           }
-        });
+        };
       }
       else {
-        return Promise.resolve({ data: {} });
+        return { data: {} };
       }
     });
-    sinon.stub(global, 'setTimeout').callsFake((fn) => {
-      fn();
-      return {} as any;
-    });
 
-    _request
+    await _request
       .get({
         url: 'https://contoso.sharepoint.com/'
-      })
-      .then(() => {
-        try {
-          assert.strictEqual(i, 4);
-          done();
-        }
-        catch (err) {
-          done(err);
-        }
-      }, (err: any) => {
-        done(err);
       });
+
+    assert.strictEqual(i, 4);
   });
 
-  it('correctly handles request that was first 429-throttled and then failed', (done) => {
+  it('correctly handles request that was first 429-throttled and then failed', async () => {
     let i: number = 0;
 
     sinon.stub(_request as any, 'req').callsFake(() => {
       if (i++ === 0) {
-        return Promise.reject({
+        throw {
           response: {
             status: 429,
             headers: {}
           }
-        });
+        };
       }
       else {
-        return Promise.reject('Error');
+        throw 'Error';
       }
     });
-    sinon.stub(global, 'setTimeout').callsFake((fn) => {
-      fn();
-      return {} as any;
-    });
 
-    _request
-      .get({
-        url: 'https://contoso.sharepoint.com/'
-      })
-      .then(() => {
-        done('Expected error');
-      }, (err) => {
-        try {
-          assert.strictEqual(err, 'Error');
-          done();
-        }
-        catch (e) {
-          done(e);
-        }
-      });
+    try {
+      await _request
+        .get({
+          url: 'https://contoso.sharepoint.com/'
+        });
+      assert.fail('Error expected');
+    }
+    catch (err) {
+      assert.strictEqual(err, 'Error');
+    }
   });
 
-  it('logs additional info for throttled requests in debug mode', (done) => {
+  it('logs additional info for throttled requests in debug mode', async () => {
     let i: number = 0;
     _request.debug = true;
     const logSpy: sinon.SinonSpy = sinon.spy(logger, 'log');
 
     sinon.stub(_request as any, 'req').callsFake(() => {
       if (i++ === 0) {
-        return Promise.reject({
+        throw {
           response: {
             status: 429,
             headers: {
               'retry-after': 10
             }
           }
-        });
+        };
       }
       else {
-        return Promise.resolve({ data: {} });
+        return { data: {} };
       }
     });
```

Comment:
The original file used to have a setTimeout stub here. Why is it missing now?

Reply:
Because we did set the timeout in the before to 0ms, which means that it would immediately complete. Using the util however we do need this.